WalletWasabi.Gui/Tabs/WalletManager/RecoverWalletViewModel.cs

```diff
@@ -46,17 +49,34 @@ public RecoverWalletViewModel(WalletManagerViewModel owner) : base("Recover Wall
 				}
 				else if (string.IsNullOrWhiteSpace(MnemonicWords))
 				{
-					ValidationMessage = $"Recovery Words were not supplied.";
+					ValidationMessage = "Recovery Words were not supplied.";
+				}
+				else if (string.IsNullOrWhiteSpace(AccountKeyPath))
+				{
+					ValidationMessage = "The account key path is not valid.";
+				}
+				else if (MinGapLimit < KeyManager.AbsoluteMinGapLimit - 1)
+				{
+					ValidationMessage = $"Min Gap Limit cannot be smaller than {KeyManager.AbsoluteMinGapLimit - 1}.";
+				}
+				else if (MinGapLimit > KeyManager.AbsoluteMaxGapLimit)
+				{
+					ValidationMessage = $"Min Gap Limit cannot be larger than {KeyManager.AbsoluteMaxGapLimit}.";
 				}
 				else
 				{
 					try
 					{
+						KeyPath keyPath = KeyPath.Parse(AccountKeyPath);
 						var mnemonic = new Mnemonic(MnemonicWords);
-						KeyManager.Recover(mnemonic, Password, walletFilePath);
+						KeyManager.Recover(mnemonic, Password, walletFilePath, keyPath, MinGapLimit + 1);
 
 						owner.SelectLoadWallet();
 					}
+					catch (FormatException)
+					{
+						ValidationMessage = "The account key path is not a valid derivation path.";
```

Comment:
The error message is misleading. Here we can get a `FormatException` for several reasons, not just for invalid derivation path. For example, if the number of mnemonics is wrong you will get this exception too and the user will review the derivation path.

My recommendation is to create a `bool TryParseKeyPath(string, KeyPath)` and use that for the moment. Next, wrap the `Recover` around a try-catch(Exception) block.